File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortedSquares = function(nums) {
    const result = [];
    let low = 0,
        high = nums.length - 1;
    while(low <= high){
        const left = nums[low]**2,
              right = nums[high]**2;
        if(left > right){
            result.push(left);
            low++
        }
        else {
            result.push(right)
            high--;
        }
    }
    return result.reverse();
};

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def sorted_squares(nums)
    
new_array = []
    
nums.each do |number|
   new_array << number**2
end 
    
new_array.sort {|a,b| a <=> b}
    
    
end